commit 7393f47bd01b53394efd89aac720a6849372d99f
Author: John Johansen <john.johansen@canonical.com>
Date:   Wed Sep 27 00:45:16 2017 -0400

    UBUNTU: SAUCE: LSM stacking: inherit current display LSM
    
    If a current display LSM is set it should be inherited. As per 2017
    LSS discussion.
    
    Signed-off-by: John Johansen <john.johansen@canonical.com>
    Signed-off-by: Seth Forshee <seth.forshee@canonical.com>

diff --git a/security/security.c b/security/security.c
index 2c118ec11f2b..e5d5213ef8d7 100644
--- a/security/security.c
+++ b/security/security.c
@@ -452,6 +452,12 @@ int lsm_task_alloc(struct task_struct *task)
 	task->security = kzalloc(blob_sizes.lbs_task, GFP_KERNEL);
 	if (task->security == NULL)
 		return -ENOMEM;
+
+	/* inherit current display lsm */
+#ifdef CONFIG_SECURITY_STACKING
+	if (current->security)
+		strcpy(task->security, lsm_of_task(current));
+#endif
 	return 0;
 }
 
